samcli/commands/init/init_templates.py

```diff
@@ -87,13 +88,16 @@ def clone_templates_repo(self):
             os_name = system().lower()
             cloned_folder_name = APP_TEMPLATES_REPO_NAME_WINDOWS if os_name == "windows" else APP_TEMPLATES_REPO_NAME
 
+            to_upsert = self._check_upsert_templates(shared_dir, cloned_folder_name)
+
             try:
-                self._git_repo.clone(
-                    clone_dir=shared_dir,
-                    clone_name=cloned_folder_name,
-                    replace_existing=True,
-                    commit=APP_TEMPLATES_REPO_COMMIT,
-                )
+                if to_upsert:
+                    self._git_repo.clone(
```

Comment:
nit: too many indentations here, any possibility to early exit and reduce them?

Reply:
so if `to_upsert` is `False`, nothing will happen in the try block?

Reply:
Yeah, we can just avoid the try block all together if we don't need to be upserting the app templates repo